src/boss/cloudvm/worker.go

```diff
@@ -118,10 +123,19 @@ func (pool *WorkerPool) startNewWorker() {
 
 	go func() { // should be able to create multiple instances simultaneously
 		worker.numTask = 1
-		pool.CreateInstance(worker) //create new instance
-
-		if pool.platform != "mock" {
+		err := pool.CreateInstance(worker) //create new instance
+		// TODO: need to handle this error, not panic (may use channel?)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+		if pool.platform == "gcp" {
```

Comment:
why do we need different logic for these two?  Can we handle errors consistently?